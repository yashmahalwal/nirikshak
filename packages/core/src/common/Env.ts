/* istanbul ignore file */
/* eslint-disable prefer-const */
export let MAX_PLURAL_ENTRIES = 10;
export let MIN_PLURAL_ENTRIES = 0;
export let RANDOMNESS_ITERATIONS = 10;
