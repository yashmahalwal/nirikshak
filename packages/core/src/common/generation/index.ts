export { generateFaker } from "./fakerGen";
export { generateCustom } from "./customGen";
