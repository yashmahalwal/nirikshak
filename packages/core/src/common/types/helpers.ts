import { CustomFunction } from "./custom";
import { Literal, isLiteral } from "./literals";

export interface SchemaHelpers {
    [key: string]: CustomFunction;
}

// Ultimately, a resource fields ends in Primitive type
// Ex: {age: 23}
// Ex: {address: [{zipCode: "370445"}, {zipCode: "121342"}], classes: [1,2,3,4,5]}
export type Primitives = Literal | Array<Primitives>;
export function isPrimitives(input: any): input is Primitives {
    if (Array.isArray(input)) return input.every((i) => isPrimitives(i));
    return isLiteral(input);
}
