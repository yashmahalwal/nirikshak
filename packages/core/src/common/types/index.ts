export { SchemaHelpers } from "./helpers";
export { isCustomFunction, CustomFunction } from "./custom";
export { isFakerType } from "./fakerTypes";
