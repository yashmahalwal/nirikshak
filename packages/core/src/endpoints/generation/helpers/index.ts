export { generateBodyType } from "./bodyTypeGen";
export { generateHeaderAndStatus } from "../output/headerAndStatus";
