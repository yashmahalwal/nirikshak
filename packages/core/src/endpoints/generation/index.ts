export { generateURL } from "./urlStringGen";
export { getFromResource } from "./resourceStringGen";
export * from "./helpers";
export * from "./input";
