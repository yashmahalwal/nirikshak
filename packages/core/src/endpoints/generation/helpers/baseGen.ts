import { Base } from "../../types/helpers";
import { ResourceInstance } from "../../../resource/types/helper";
import { SchemaHelpers, Primitives } from "../../../common/types/helpers";
import { generateBaseType } from "./baseTypeGen";

export function generateBase(
    input: Base,
    instance: ResourceInstance,
    helpers: SchemaHelpers
): Promise<Primitives> {
    if (Array.isArray(input)) {
        return Promise.all(
            (input as Base[]).map((entry) =>
                generateBase(entry, instance, helpers)
            )
        );
    }

    return generateBaseType(input, instance, helpers);
}
