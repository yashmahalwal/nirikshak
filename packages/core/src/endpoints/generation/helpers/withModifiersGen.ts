import {
    WithModifiers,
    BodyType,
    Base,
    isWithModifiersBase,
    BodyInstance,
} from "../../types/helpers";
import { ResourceInstance } from "../../../resource/types/helper";
import { SchemaHelpers, Primitives } from "../../../common/types/helpers";
import faker from "faker";
import { MIN_PLURAL_ENTRIES, MAX_PLURAL_ENTRIES } from "../../../common/Env";
import { generateBase } from "./baseGen";
import { generateBodyType } from "./bodyTypeGen";

export async function generateWithModifiersBase(
    input: WithModifiers<Base>,
    resource: ResourceInstance,
    helpers: SchemaHelpers
): Promise<Primitives> {
    if (input.nullable && faker.random.boolean()) return null;

    if (input.plural) {
        const arr: Promise<Primitives>[] = [];

        const length = faker.random.number({
            min: MIN_PLURAL_ENTRIES,
            max: MAX_PLURAL_ENTRIES,
            precision: 1,
        });

        for (let i = 0; i < length; i++)
            arr.push(generateBase(input.type, resource, helpers));

        return Promise.all(arr);
    }

    return generateBase(input.type, resource, helpers);
}

export async function generateWithModifiersBodyType(
    input: WithModifiers<BodyType>,
    resource: ResourceInstance,
    helpers: SchemaHelpers
): Promise<BodyInstance | BodyInstance[] | null> {
    if (input.nullable && faker.random.boolean()) return null;

    if (input.plural) {
        const arr: Promise<BodyInstance>[] = [];

        const length = faker.random.number({
            min: MIN_PLURAL_ENTRIES,
            max: MAX_PLURAL_ENTRIES,
            precision: 1,
        });

        for (let i = 0; i < length; i++)
            arr.push(generateBodyType(input.field, resource, helpers));

        return Promise.all(arr);
    }

    return generateBodyType(input.field, resource, helpers);
}

export async function generateWithModifiers<T extends BodyType | Base>(
    input: WithModifiers<T>,
    resource: ResourceInstance,
    helpers: SchemaHelpers
): Promise<
    null | (T extends Base ? Primitives : BodyInstance[] | BodyInstance)
> {
    if (isWithModifiersBase(input))
        return generateWithModifiersBase(input, resource, helpers) as any;
    return generateWithModifiersBodyType(
        input as WithModifiers<BodyType>,
        resource,
        helpers
    ) as any;
}
