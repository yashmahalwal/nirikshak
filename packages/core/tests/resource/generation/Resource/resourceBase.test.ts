/* 
    Test to validate resource generation
    Not testing in depth as other functions are assumed to be working correctly
*/
import {
    ResourceBase,
    Resource,
} from "../../../../src/resource/types/resource";
import { RANDOMNESS_ITERATIONS } from "../../../../src/common/Env";
import {
    generateResourceBase,
    generateResource,
} from "../../../../src/resource/generation/resourceGen";
import faker from "faker";
import { SchemaHelpers } from "../../../../src/common/types/helpers";
import {
    isWithModifiersResource,
    ResourceInstanceBase,
} from "../../../../src/resource/types/helper";
import { generateWithModifiers } from "../../../../src/resource/generation/withModifiersGen";
import {
    generateOneOfEntries,
    SelectedType,
} from "../../../../src/resource/generation/oneOfEntries";
import {
    isWithModifiersBase,
    BaseType,
    WithModifiers,
} from "../../../../src/endpoints/types/helpers";
import { generateBaseType } from "../../../../src/resource/generation/baseGen";
// Helpers for the resource
const Helpers: SchemaHelpers = {
    username: async () => faker.name.firstName() + faker.name.lastName(),
    number: async (min: number, max: number, step: number) =>
        // Use faker.random.number instead of Math.random
        // For universal randomness seed
        faker.random.number({ min, max, precision: step }),
    async invalidHelper() {
        return { age: 12 } as any;
    },
};

const ValidResourceBases: {
    input: ResourceBase;
    output: () => Promise<ResourceInstanceBase>;
}[] = [
    {
        input: {
            name: 15,
            class: "faker:random.number",
            branch: { function: "faker:random.words", args: [1] },
            address: "faker.random.zipCode",
        },
        output: async (): Promise<ResourceInstanceBase> => {
            return {
                name: 15,
                class: faker.random.number(),
                branch: faker.random.words(1),
                address: "faker.random.zipCode",
            };
        },
    },
    {
        input: {
            root: {
                f2: {
                    type: [1, 2, 3, 4, "stirng", false, "faker:random.address"],
                    nullable: true,
                    plural: true,
                    optional: true,
                },
                f1: {
                    field: {
                        f3: {
                            fields: [
                                {
                                    branch: {
                                        types: ["CSE", "ECE", "Mechanical"],
                                        plural: false,
                                        nullable: false,
                                        optional: true,
                                    },
                                    section: {
                                        function: "custom:username",
                                        args: ["CSE", "ECE", "Mechanical"],
                                    },
                                },
                                {
                                    addresses: {
                                        field: {
                                            city: {
                                                types: [
                                                    "faker:random.words",
                                                    "faker:address.city",
                                                ],
                                            },
                                            street: {
                                                fields: [
                                                    {
                                                        streetName:
                                                            "faker:address.streetName",
                                                    },
                                                    {
                                                        streetAddress:
                                                            "faker:address.streetAddress",
                                                    },
                                                ],
                                                types: [
                                                    "faker:address.zipCode",
                                                ],
                                            },
                                        },
                                        plural: true,
                                        optional: true,
                                        nullable: false,
                                    },
                                },
                            ],
                            types: [
                                {
                                    type: {
                                        function: "custom:number",
                                        args: [1, 7, 0.2],
                                    },
                                    plural: true,
                                    optional: true,
                                    nullable: false,
                                },
                                null,
                                "faker:random.number",
                                {
                                    function: "faker:lorem.lines",
                                },
                                17.885,
                                false,
                                true,
                            ],
                        },
                    },
                    optional: true,
                },
            },
        },
        output: async function (): Promise<ResourceInstanceBase> {
            const o: Record<string, any> = {
                root: {},
            };

            for (const key in this.input.root) {
                switch (key) {
                    case "f2":
                        if (!faker.random.boolean())
                            // Optional
                            o.root["f2"] = await generateWithModifiers(
                                {
                                    type: [
                                        1,
                                        2,
                                        3,
                                        4,
                                        "stirng",
                                        false,
                                        "faker:random.address",
                                    ],
                                    nullable: true,
                                    plural: true,
                                    optional: true,
                                },
                                Helpers
                            );
                        break;
                    case "f1":
                        if (!faker.random.boolean()) {
                            // Optional
                            o.root["f1"] = {};
                            const entry = generateOneOfEntries(
                                this.input.root.f1.field.f3
                            );

                            if (entry.type === "base type") {
                                if (isWithModifiersBase(entry.selection)) {
                                    o.root.f1.f3 = await generateWithModifiers(
                                        entry.selection,
                                        Helpers
                                    );
                                } else {
                                    o.root.f1.f3 = await generateBaseType(
                                        entry.selection,
                                        Helpers
                                    );
                                }
                            } else {
                                if (isWithModifiersResource(entry.selection)) {
                                    o.root.f1.f3 = await generateWithModifiers(
                                        entry.selection,
                                        Helpers
                                    );
                                } else {
                                    o.root.f1.f3 = await generateResourceBase(
                                        entry.selection,
                                        Helpers
                                    );
                                }
                            }
                        }
                        break;
                }
            }
            return o;
        },
    },
    {
        input: {
            f1: {
                types: [
                    1,
                    2,
                    3,
                    ["stringify", 19, 30, 19.542],
                    "faker:random.number",
                    [[[[["custom.username"]]]]],
                ],
            },
            f2: {
                fields: [
                    { branch: "CSE" },
                    {
                        age: 12,
                        name: 14,
                        address: {
                            types: [
                                12.4353,
                                "faker:random.number",
                                { type: true, nullable: true },
                                [12, 13, 14, null, false, "myString"],
                            ],
                        },
                    },
                ],
            },
            f3: {
                fields: [
                    { branch: 3 },
                    {
                        age: {
                            type: {
                                function: "faker:random.number",
                                args: [{ min: 1, max: 100, precision: 0.1 }],
                            },
                            nullable: true,
                        },
                    },
                ],
                types: [12, 14, true, true, false, ["SIMP"]],
            },
            f4: {
                fields: [
                    {
                        field: { name: "Name", surName: "Surname" },
                        plural: true,
                    },
                ],
            },
        },
        output: async function (): Promise<ResourceInstanceBase> {
            const o = {};

            for (const key in this.input) {
                switch (key) {
                    case "f1":
                        {
                            const f1Entry = generateOneOfEntries(
                                this.input.f1
                            ) as SelectedType<BaseType>;
                            o["f1"] = await generateBaseType(
                                f1Entry.selection as BaseType,
                                Helpers
                            );
                        }
                        break;
                    case "f2":
                        {
                            const f2Entry = generateOneOfEntries(
                                this.input.f2
                            ) as SelectedType<ResourceBase>;
                            o["f2"] = await generateResourceBase(
                                f2Entry.selection as ResourceBase,
                                Helpers
                            );
                        }
                        break;
                    case "f3":
                        {
                            const f3Entry = generateOneOfEntries(this.input.f3);

                            if (f3Entry.type === "base type") {
                                o["f3"] = await generateBaseType(
                                    f3Entry.selection as BaseType,
                                    Helpers
                                );
                            } else {
                                o["f3"] = await generateResourceBase(
                                    f3Entry.selection as ResourceBase,
                                    Helpers
                                );
                            }
                        }
                        break;
                    case "f4":
                        {
                            const f4Entry = generateOneOfEntries(
                                this.input.f4
                            ) as SelectedType<ResourceBase>;
                            o["f4"] = await generateWithModifiers(
                                f4Entry.selection as WithModifiers<
                                    ResourceBase
                                >,
                                Helpers
                            );
                        }
                        break;
                }
            }

            return o;
        },
    },
];

const InvalidResourceBases = [
    {
        name: "faker:name.firstName",
        age: "custom:age",
    },
    { name: "faker:name.lastName", age: "custom:invalidHelper" },
];

describe(`Resource base type`, () => {
    // Account for randomness
    for (let i = 0; i < RANDOMNESS_ITERATIONS; i++)
        ValidResourceBases.forEach((resource, index) =>
            test(`resource base: ${index}, iteration: ${i}`, async () => {
                faker.seed(i + 123);
                const o = Object.assign({}, resource.input) as Resource;
                const got = await generateResourceBase(o, Helpers);
                faker.seed(i + 123);
                const expected = await resource.output();
                expect(got).toEqual(expected);
            })
        );
});
describe(`Resource type`, () => {
    beforeAll(() => faker.seed(123));
    // Account for randomness
    for (let i = 0; i < RANDOMNESS_ITERATIONS; i++)
        ValidResourceBases.forEach((resource, index) =>
            test(`resource base ${index}, iteration: ${i}`, async () => {
                const o = Object.assign({}, resource.input) as Resource;
                o.id = "faker:random.word";
                expect(await generateResource(o, Helpers)).toMatchSnapshot();
            })
        );

    test.each(InvalidResourceBases)(`Invalid resource base: %#`, (entry) =>
        expect(
            generateResourceBase(entry as ResourceBase, Helpers)
        ).rejects.toMatchSnapshot()
    );
});
