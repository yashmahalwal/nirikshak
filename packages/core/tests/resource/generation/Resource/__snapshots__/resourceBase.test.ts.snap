// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Resource type Invalid resource base: 0 1`] = `[Error: function: custom:age, args: ]`;

exports[`Resource type Invalid resource base: 1 1`] = `[Error: Non literal type value from custom:invalidHelper]`;

exports[`Resource type resource base 0, iteration: 0 1`] = `
Object {
  "address": "faker.random.zipCode",
  "branch": "port",
  "class": 69646,
  "id": "Rufiyaa",
  "name": 15,
}
`;

exports[`Resource type resource base 0, iteration: 1 1`] = `
Object {
  "address": "faker.random.zipCode",
  "branch": "Handcrafted Cotton Soap",
  "class": 24475,
  "id": "Table",
  "name": 15,
}
`;

exports[`Resource type resource base 0, iteration: 2 1`] = `
Object {
  "address": "faker.random.zipCode",
  "branch": "Enterprise-wide",
  "class": 37380,
  "id": "invoice",
  "name": 15,
}
`;

exports[`Resource type resource base 0, iteration: 3 1`] = `
Object {
  "address": "faker.random.zipCode",
  "branch": "Configurable",
  "class": 91669,
  "id": "Manager",
  "name": 15,
}
`;

exports[`Resource type resource base 0, iteration: 4 1`] = `
Object {
  "address": "faker.random.zipCode",
  "branch": "deliver",
  "class": 38017,
  "id": "Supervisor",
  "name": 15,
}
`;

exports[`Resource type resource base 0, iteration: 5 1`] = `
Object {
  "address": "faker.random.zipCode",
  "branch": "Gorgeous",
  "class": 55204,
  "id": "hub",
  "name": 15,
}
`;

exports[`Resource type resource base 0, iteration: 6 1`] = `
Object {
  "address": "faker.random.zipCode",
  "branch": "Gabon",
  "class": 58063,
  "id": "Technician",
  "name": 15,
}
`;

exports[`Resource type resource base 0, iteration: 7 1`] = `
Object {
  "address": "faker.random.zipCode",
  "branch": "Unions",
  "class": 88694,
  "id": "Small",
  "name": 15,
}
`;

exports[`Resource type resource base 0, iteration: 8 1`] = `
Object {
  "address": "faker.random.zipCode",
  "branch": "needs-based",
  "class": 81103,
  "id": "withdrawal",
  "name": 15,
}
`;

exports[`Resource type resource base 0, iteration: 9 1`] = `
Object {
  "address": "faker.random.zipCode",
  "branch": "Brand",
  "class": 17151,
  "id": "Kentucky",
  "name": 15,
}
`;

exports[`Resource type resource base 1, iteration: 0 1`] = `
Object {
  "id": "mobile",
  "root": Object {},
}
`;

exports[`Resource type resource base 1, iteration: 1 1`] = `
Object {
  "id": "withdrawal",
  "root": Object {
    "f2": Array [
      Array [
        1,
        2,
        3,
        4,
        "stirng",
        false,
        "faker:random.address",
      ],
      Array [
        1,
        2,
        3,
        4,
        "stirng",
        false,
        "faker:random.address",
      ],
      Array [
        1,
        2,
        3,
        4,
        "stirng",
        false,
        "faker:random.address",
      ],
      Array [
        1,
        2,
        3,
        4,
        "stirng",
        false,
        "faker:random.address",
      ],
      Array [
        1,
        2,
        3,
        4,
        "stirng",
        false,
        "faker:random.address",
      ],
    ],
  },
}
`;

exports[`Resource type resource base 1, iteration: 2 1`] = `
Object {
  "id": "Palladium",
  "root": Object {
    "f1": Object {
      "f3": "Sequi molestiae beatae enim necessitatibus molestiae.",
    },
    "f2": null,
  },
}
`;

exports[`Resource type resource base 1, iteration: 3 1`] = `
Object {
  "id": "Cuba",
  "root": Object {
    "f1": Object {
      "f3": Object {
        "addresses": Array [
          Object {
            "city": "Millsburgh",
            "street": Object {
              "streetName": "Ronny Ramp",
            },
          },
          Object {
            "city": "West Adelia",
            "street": "78211",
          },
          Object {
            "city": "infomediaries Vatu",
            "street": "05816",
          },
          Object {
            "city": "South Jaclyn",
            "street": Object {
              "streetAddress": "045 Joel Groves",
            },
          },
          Object {
            "city": "New Rodger",
            "street": "76135",
          },
        ],
      },
    },
  },
}
`;

exports[`Resource type resource base 1, iteration: 4 1`] = `
Object {
  "id": "GB",
  "root": Object {
    "f2": null,
  },
}
`;

exports[`Resource type resource base 1, iteration: 5 1`] = `
Object {
  "id": "Hong Kong Dollar",
  "root": Object {
    "f1": Object {
      "f3": Object {
        "branch": "ECE",
        "section": "RebecaSteuber",
      },
    },
  },
}
`;

exports[`Resource type resource base 1, iteration: 6 1`] = `
Object {
  "id": "bus",
  "root": Object {
    "f1": Object {
      "f3": "Occaecati veniam ut exercitationem voluptatibus error.
Porro error sint aut aliquid.",
    },
  },
}
`;

exports[`Resource type resource base 1, iteration: 7 1`] = `
Object {
  "id": "Practical Steel Cheese",
  "root": Object {
    "f2": null,
  },
}
`;

exports[`Resource type resource base 1, iteration: 8 1`] = `
Object {
  "id": "Table",
  "root": Object {
    "f1": Object {
      "f3": Object {},
    },
    "f2": Array [
      Array [
        1,
        2,
        3,
        4,
        "stirng",
        false,
        "faker:random.address",
      ],
      Array [
        1,
        2,
        3,
        4,
        "stirng",
        false,
        "faker:random.address",
      ],
      Array [
        1,
        2,
        3,
        4,
        "stirng",
        false,
        "faker:random.address",
      ],
      Array [
        1,
        2,
        3,
        4,
        "stirng",
        false,
        "faker:random.address",
      ],
      Array [
        1,
        2,
        3,
        4,
        "stirng",
        false,
        "faker:random.address",
      ],
      Array [
        1,
        2,
        3,
        4,
        "stirng",
        false,
        "faker:random.address",
      ],
      Array [
        1,
        2,
        3,
        4,
        "stirng",
        false,
        "faker:random.address",
      ],
    ],
  },
}
`;

exports[`Resource type resource base 1, iteration: 9 1`] = `
Object {
  "id": "generating",
  "root": Object {
    "f1": Object {
      "f3": 17.885,
    },
  },
}
`;

exports[`Resource type resource base 2, iteration: 0 1`] = `
Object {
  "f1": 3,
  "f2": Object {
    "address": true,
    "age": 12,
    "name": 14,
  },
  "f3": true,
  "f4": Array [
    Object {
      "name": "Name",
      "surName": "Surname",
    },
    Object {
      "name": "Name",
      "surName": "Surname",
    },
    Object {
      "name": "Name",
      "surName": "Surname",
    },
  ],
  "id": "help-desk",
}
`;

exports[`Resource type resource base 2, iteration: 1 1`] = `
Object {
  "f1": Array [
    Array [
      Array [
        Array [
          Array [
            "custom.username",
          ],
        ],
      ],
    ],
  ],
  "f2": Object {
    "branch": "CSE",
  },
  "f3": Object {
    "age": null,
  },
  "f4": Array [
    Object {
      "name": "Name",
      "surName": "Surname",
    },
    Object {
      "name": "Name",
      "surName": "Surname",
    },
    Object {
      "name": "Name",
      "surName": "Surname",
    },
    Object {
      "name": "Name",
      "surName": "Surname",
    },
    Object {
      "name": "Name",
      "surName": "Surname",
    },
    Object {
      "name": "Name",
      "surName": "Surname",
    },
    Object {
      "name": "Name",
      "surName": "Surname",
    },
  ],
  "id": "withdrawal",
}
`;

exports[`Resource type resource base 2, iteration: 2 1`] = `
Object {
  "f1": Array [
    "stringify",
    19,
    30,
    19.542,
  ],
  "f2": Object {
    "address": 62395,
    "age": 12,
    "name": 14,
  },
  "f3": Object {
    "branch": 3,
  },
  "f4": Array [
    Object {
      "name": "Name",
      "surName": "Surname",
    },
    Object {
      "name": "Name",
      "surName": "Surname",
    },
    Object {
      "name": "Name",
      "surName": "Surname",
    },
  ],
  "id": "Vatu",
}
`;

exports[`Resource type resource base 2, iteration: 3 1`] = `
Object {
  "f1": 3,
  "f2": Object {
    "branch": "CSE",
  },
  "f3": Object {
    "age": null,
  },
  "f4": Array [
    Object {
      "name": "Name",
      "surName": "Surname",
    },
    Object {
      "name": "Name",
      "surName": "Surname",
    },
    Object {
      "name": "Name",
      "surName": "Surname",
    },
  ],
  "id": "Front-line",
}
`;

exports[`Resource type resource base 2, iteration: 4 1`] = `
Object {
  "f1": 2,
  "f2": Object {
    "branch": "CSE",
  },
  "f3": Object {
    "branch": 3,
  },
  "f4": Array [
    Object {
      "name": "Name",
      "surName": "Surname",
    },
    Object {
      "name": "Name",
      "surName": "Surname",
    },
    Object {
      "name": "Name",
      "surName": "Surname",
    },
  ],
  "id": "Baby",
}
`;

exports[`Resource type resource base 2, iteration: 5 1`] = `
Object {
  "f1": 35426,
  "f2": Object {
    "address": 12.4353,
    "age": 12,
    "name": 14,
  },
  "f3": false,
  "f4": Array [
    Object {
      "name": "Name",
      "surName": "Surname",
    },
    Object {
      "name": "Name",
      "surName": "Surname",
    },
    Object {
      "name": "Name",
      "surName": "Surname",
    },
  ],
  "id": "Moldova",
}
`;

exports[`Resource type resource base 2, iteration: 6 1`] = `
Object {
  "f1": 1,
  "f2": Object {
    "branch": "CSE",
  },
  "f3": Object {
    "age": null,
  },
  "f4": Array [
    Object {
      "name": "Name",
      "surName": "Surname",
    },
    Object {
      "name": "Name",
      "surName": "Surname",
    },
    Object {
      "name": "Name",
      "surName": "Surname",
    },
    Object {
      "name": "Name",
      "surName": "Surname",
    },
    Object {
      "name": "Name",
      "surName": "Surname",
    },
    Object {
      "name": "Name",
      "surName": "Surname",
    },
    Object {
      "name": "Name",
      "surName": "Surname",
    },
    Object {
      "name": "Name",
      "surName": "Surname",
    },
    Object {
      "name": "Name",
      "surName": "Surname",
    },
    Object {
      "name": "Name",
      "surName": "Surname",
    },
  ],
  "id": "Practical Steel Mouse",
}
`;

exports[`Resource type resource base 2, iteration: 7 1`] = `
Object {
  "f1": 3,
  "f2": Object {
    "address": true,
    "age": 12,
    "name": 14,
  },
  "f3": false,
  "f4": Array [
    Object {
      "name": "Name",
      "surName": "Surname",
    },
    Object {
      "name": "Name",
      "surName": "Surname",
    },
    Object {
      "name": "Name",
      "surName": "Surname",
    },
    Object {
      "name": "Name",
      "surName": "Surname",
    },
    Object {
      "name": "Name",
      "surName": "Surname",
    },
    Object {
      "name": "Name",
      "surName": "Surname",
    },
    Object {
      "name": "Name",
      "surName": "Surname",
    },
    Object {
      "name": "Name",
      "surName": "Surname",
    },
    Object {
      "name": "Name",
      "surName": "Surname",
    },
  ],
  "id": "neural",
}
`;

exports[`Resource type resource base 2, iteration: 8 1`] = `
Object {
  "f1": 1,
  "f2": Object {
    "address": null,
    "age": 12,
    "name": 14,
  },
  "f3": 12,
  "f4": Array [
    Object {
      "name": "Name",
      "surName": "Surname",
    },
    Object {
      "name": "Name",
      "surName": "Surname",
    },
    Object {
      "name": "Name",
      "surName": "Surname",
    },
    Object {
      "name": "Name",
      "surName": "Surname",
    },
    Object {
      "name": "Name",
      "surName": "Surname",
    },
    Object {
      "name": "Name",
      "surName": "Surname",
    },
  ],
  "id": "Hill",
}
`;

exports[`Resource type resource base 2, iteration: 9 1`] = `
Object {
  "f1": Array [
    Array [
      Array [
        Array [
          Array [
            "custom.username",
          ],
        ],
      ],
    ],
  ],
  "f2": Object {
    "address": 12.4353,
    "age": 12,
    "name": 14,
  },
  "f3": 12,
  "f4": Array [
    Object {
      "name": "Name",
      "surName": "Surname",
    },
  ],
  "id": "Practical",
}
`;
